collections/nemo_nlp/nemo_nlp/utils/callbacks/token_classification.py

```diff
@@ -71,6 +71,12 @@ def eval_epochs_done_callback(global_vars,
         i = random.randint(0, preds.shape[0] - sample_size - 1)
     logger.info("Sampled preds: [%s]" % list2str(preds[i:i+sample_size]))
     logger.info("Sampled labels: [%s]" % list2str(labels[i:i+sample_size]))
+
+    # remove labels from label_ids that don't appear in the dev set
+    missing_labels = set(label_ids.values()) - set(labels) - set(preds)
+    label_ids = {k: label_ids[k]
+                 for k, v in label_ids.items() if v not in missing_labels}
+
```

Comment:
the logic is slightly backward. i'd write something like this:
used_labels = set(labels) + set(preds)
label_ids = {k: label_ids[k]
                 for k, v in label_ids.items() if v in used_labels}